src/System.Security.Cryptography.Cng/src/System/Security/Cryptography/CngKey.EC.cs

```diff
@@ -0,0 +1,185 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Internal.Cryptography;
+using System.Diagnostics;
+using BCRYPT_ECC_PARAMETER_HEADER = Interop.BCrypt.BCRYPT_ECC_PARAMETER_HEADER;
+
+namespace System.Security.Cryptography
+{
+    public sealed partial class CngKey : IDisposable
+    {
+        /// <summary>
+        /// Does the key use elliptic curve cryptography
+        /// </summary>
+        /// <returns></returns>
+        internal bool IsEcc()
+        {
+            return (AlgorithmGroup == CngAlgorithmGroup.ECDiffieHellman ||
+                AlgorithmGroup == CngAlgorithmGroup.ECDsa);
+        }
+
+        /// <summary>
+        /// Does the key represent a named curve (Win10+)
+        /// </summary>
+        /// <returns></returns>
+        internal bool IsECNamedCurve()
+        {
+            return IsECNamedCurve(Algorithm.Algorithm);
+        }
+
+        internal static bool IsECNamedCurve(string algorithm)
+        {
+            return (algorithm == CngAlgorithm.ECDiffieHellman.Algorithm ||
+                algorithm == CngAlgorithm.ECDsa.Algorithm);
+        }
+
+        internal string GetCurveName()
+        {
+            if (IsECNamedCurve())
+            {
+                // Win10+
```

Comment:
We don't really know it's 10. I tweaked the comment. The named curves use a generic algorithm type ECDsa or ECDH which are only supported on 10+. Non-named curves use specific algorithm type for each curve type, so ECDsaP256 for example which are supported on <10. So this method just wraps that check (since we want to support <10 for certain APIs)